src/zjs_net.c

```diff
@@ -259,13 +259,12 @@ static void tcp_received(struct net_context *context,
 {
     sock_handle_t *handle = (sock_handle_t *)user_data;
 
-    if (handle) {
+    if (handle && buf) {
         start_socket_timeout(handle, handle->timeout);
 
         uint32_t len = net_nbuf_appdatalen(buf);
         uint8_t *data = net_nbuf_appdata(buf);
 
-        DBG_PRINT("received data, context=%p, data=%p, len=%u\n", context, data, len);
         // TODO: Probably not the right way to check if a socket is closed
```

Comment:
@jprestwo : I don't think there's any meaning in 2,3,4, these are just artifacts of 1.7 implementation, as I mentioned, these are gone in Z master (to be 1.8). A lot of things changes in 1.8, the patch will be all red and green, and important thing to keep in mind, it's not just mechanical search&replace.

buf == NULL is officially documented though:

~~~
 * @param pkt Network buffer that is received. If the pkt is not NULL,
 * then the callback will own the buffer and it needs to to unref the pkt
 * as soon as it has finished working with it.  On EOF, pkt will be NULL.
 * @param status Value is set to 0 if some data or the connection is
 * at EOF, <0 if there was an error receiving data, in this case the
 * pkt parameter is set to NULL.
~~~